### ISSUE TITLE
When are we gonna start a RL training session?

### ISSUE THREAD

**Issue by Error323** [glinscott/leela-chess#44]
I was just wondering, what would be a good milestone to start training a small network tabula rasa? How big will it be? 64x5? I would like to work towards that goal!

- Obviously we need a distributed management server/client system.
- Will we go for an Alpha Zero structure or AlphaGo Zero? (I think the former would be better, but requires some kind of adaptive throughput on the training end.)
- A webpage showing results

I'm gathering every GPU system I can find :smirk: I can't wait! Perhaps we can ask for sponsors through GM players?

**Comment by grondilu** [comment]
> Perhaps we can ask for sponsors through GM players?

Several top players have expressed interest in AlphaZero.  At least two of them (former world championship challenger Sergei Karjakine and French top player Maxime Vachier Lagrave) have estimated the price of such technology to several hundred thousand dollars.

It is not clear this interest would be the same for a public project though, as the economic value is much higher for a player if he can gain an exclusive access to the technology.  That being said, players do value the analysis of Stockfish, which is FOSS.

I would suggest publicizing leela-chess on Twitter, where most top players are active.

**Comment by glinscott** [comment]
I've got the client uploading games to the server now.  Need to make things a bit more robust and should be able to kick off the public part soon!

**Comment by kiudee** [comment]
For the _AlphaZero_ training it would be useful for us to have a dataset structure of games which is shared by the training thread and the server. The server will continually append new games, while the training thread samples random batches (or chunks, if that is more efficient) of the latest games.
Am I forgetting anything?

**Comment by Error323** [comment]
@kiudee Yeah we need to be aware of the input rate of new games in order to determine when to drop the learningrate.

**Comment by gcp** [comment]
>Yeah we need to be aware of the input rate of new games in order to determine when to drop the learningrate.

You need to drop the learning rate when the loss no longer lowers, it's not a function of the game count.

**Comment by Error323** [comment]
@gcp Indeed, that would be better! If both MSE and Policy are stalling, we can drop. I was thinking in terms of replicating the AlphaZero work, they have a steady flow of incoming games.

**Comment by Error323** [comment]
We're doing it ^_^
